iphone/cli/commands/_build.js

```diff
@@ -5261,6 +5262,22 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
 					let defaultIconChanged = false,
 						defaultIconHasAlpha = false;
 
+					// Add macOS icons if target is macOS
+					if (this.target === 'macos') {
```

Comment:
We should check for target 'dist-macappstore' as well ?
```suggestion
					if (this.target === 'macos' || this.target === 'dist-macappstore') {
```